[WEBSITE-497] Add documentation to run Jenkins on Java 11 (#2097)
jenkins-infra/jenkins.io: content/doc/administration/requirements/java.adoc

Modern Jenkins versions have the following Java requirements:

* Java 8 runtime environments, both 32-bit and 64-bit versions are supported

> * Since Jenkins `2.155`, Java 11 runtime environments is supported

Reviewer: I would put 2.163, and use a `footnote:[]` if we want to explain details why 2.155 was actually the first one. So in case users read this too quickly, they don't end up with 2.155.

Author: Noted, will do.

Author: I did that.

> * Since Jenkins `2.163` footnote:[since `2.155` but it was made easier in `2.163`], Java 11 runtime environments is supported

Author: for @MRamonLeon 

```suggestion
* Since Jenkins `2.164` footnote:[since `2.155` but it was made easier in `2.163`], Java 11 runtime environments is supported
```

Reviewer 2: To be accurate, 2.164 gather all of the simplifications, I think, although the big one was on 2.163:
```suggestion
* Since Jenkins `2.164` footnote:[since `2.155` but it was made easier in `2.164`], Java 11 runtime environments is supported
```

> * Since Jenkins `2.164` footnote:[since `2.155` but it was made easier in `2.164`], Java 11 runtime environments is supported

Reviewer: NIT: _environments_ followed by _is_ seems incorrect (?)

Author: ```suggestion
* Since Jenkins `2.164` footnote:[since `2.155` but it was made easier in `2.164`], Java 11 runtime environments are supported
```

> * Since Jenkins `2.164` footnote:[since `2.155` but it was made easier in `2.164`], Java 11 runtime environments are supported

Reviewer 3: ```suggestion
* Java 11 is supported starting from 2.164
```

We do not formally support Java 11 for previous Jenkins versions, probably we should not overcomplicate it here

Reviewer: Given this was already hinted in a footnote above, I tend to agree this may be a unnecessary repetition about the pre-2.164 story.

> ** Running Jenkins with Java 11 is documented link:/doc/administration/requirements/jenkins-on-java-11[here] 

Reviewer 4: Suggest  "** Running Jenkins with Java 11 is documented in link:/doc/administration/requirements/jenkins-on-java-11[Running Jenkins on Java 11]."

> ** in order to run Jenkins on Java 11, there are few precautions to take, which are detailed link:/doc/administration/requirements/jenkins-on-java-11[here].

Reviewer 2: To improve accessibility it's better to avoid _click here_: https://www.w3.org/QA/Tips/noClickHere

Author: is it better with this @MRamonLeon ?

```suggestion
** in order to run Jenkins on Java 11, there are few precautions to take, which are detailed in the link:/doc/administration/requirements/jenkins-on-java-11[Running Jenkins on Java 11] documentation.
```

> ** in order to run Jenkins on Java 11, there are few precautions to take, which are detailed in the link:/doc/administration/requirements/jenkins-on-java-11[Running Jenkins on Java 11] documentation.

Reviewer 3: There is no precautions on the linked page, everything is documented in the Upgrade guide

```suggestion
** Running Jenkins with Java 11 is documented link:/doc/administration/requirements/jenkins-on-java-11[here] 
```

> ** There are some precautions to take when upgrading from Java 8 to Java 11 in Jenkins, please link:/doc/administration/requirements/upgrade-java-guidelines[follow these guidelines].

Reviewer 4: Suggest "** If you are upgrading Jenkins from Java 8 to Java 11, there are some precautions you should take. Please link:/doc/administration/requirements/upgrade-java-guidelines[follow these guidelines]."

> ** if you need to upgrade the JVM version used to run Jenkins from Java 8 to Java 11, please link:/doc/administration/requirements/upgrade-java-guidelines[follow this guidelines].

Reviewer 3: ```suggestion
** There are some precautions to take when upgrading from Java 8 to Java 11 in Jenkins, please link:/doc/administration/requirements/upgrade-java-guidelines[follow these guidelines].
```
